dyno-core/src/main/java/com/netflix/dyno/connectionpool/impl/lb/HostSelectionWithFallback.java

```diff
@@ -120,9 +120,14 @@ public HostSelectionWithFallback(ConnectionPoolConfiguration config, ConnectionP
 			try { 
 				return hostPool.borrowConnection(duration, unit);
 			} catch (DynoConnectException e) {
-				lastEx = e;
-				cpMonitor.incOperationFailure(null, e);
-				useFallback = true;
+				if (e instanceof PoolExhaustedException) {
```

Comment:
We got PoolExhaustedException throwing when the pool has zero connections, if we don't do fall back immediately, when are we going to do the fall back?  Also, what happens if the pool can actually never have any connections such as in RF=2? In this RF=2 case, if we don't fall back immediately, there would be a huge performance degradation, no?